packages/actions/test/unit/security.test.ts

```diff
@@ -89,70 +110,281 @@ describe("Security", () => {
+        const clientId = githubClientId
+
+        let userId: string = ""
+
+        beforeAll(async () => signOut(userAuth))
 
-        it("should throw an error if a coordiantor tries to read another user's document", async () => {
-            // login as coordinator
-            await signInWithEmailAndPassword(userAuth, user3.data.email, user3Pwd)
-            // retrieve the document of another user
-            assert.isRejected(getDocumentById(userFirestore, commonTerms.collections.users.name, user1.uid))
+        /// @note self explanatory
+        /// one user should not be able to connect with the wrong password
+        it("should not let anyone authenticate with the wrong password", async () => {
+            const wrongPassword = "wrongPassword"
+            expect(signInWithEmailAndPassword(userAuth, users[0].data.email, wrongPassword)).to.be.rejectedWith(
+                "Firebase: Error (auth/wrong-password)."
+            )
+            expect(() => getCurrentFirebaseAuthUser(userApp)).to.throw(
+                "Unable to find the user currently authenticated with Firebase. Verify that the Firebase application is properly configured and repeat user authentication before trying again."
+            )
         })
+        if (envType === TestingEnvironment.PRODUCTION) {
+            /// @note it is not recommended to allow anynomous access to firebase
+            it("should not allow to authenticate anynomously to Firebase", async () => {
+                const auth = getAuth()
+                await expect(signInAnonymously(auth)).to.be.rejectedWith(
+                    "Firebase: Error (auth/admin-restricted-operation)."
+                )
+            })
+            /// @note it should not authenticate with a wrong OAuth2 token
+            it("should prevent authentication with the wrong OAuth2 token", async () => {
+                await expect(signInToFirebaseWithCredentials(userApp, new OAuthCredential())).to.be.rejectedWith(
+                    "Firebase: Invalid IdP response/credential: http://localhost?&providerId=undefined (auth/invalid-credential-or-provider-id)."
+                )
+            })
+            /// @note If a token has been invalidated, this shuold not allow to access Firebase again
+            /// @todo might not be able to test this in code since it requires revoking access on GitHub
+            it.skip("should not be able to authenticate with a token after this is invalidated", async () => {
+                const auth = createOAuthDeviceAuth({
+                    clientType,
+                    clientId,
+                    scopes: ["gist"],
+                    onVerification: simulateOnVerification
+                })
+                const { token } = await auth({
+                    type: tokenType
+                })
+                // Get and exchange credentials.
+                const userFirebaseCredentials = GithubAuthProvider.credential(token)
```

Comment:
How does this become invalidated for the test?

Reply:
I believe it's not possible to revoke programmatically a GitHub OAuth2 token. So this would require a manual step (or automated with puppeteer). I don't think it's worth adding this as code. However is left here as reminder because it might fit in with this: https://github.com/quadratic-funding/mpc-phase2-suite/issues/171#issuecomment-1382186016